vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/CheckboxGroup.java

```diff
@@ -484,7 +484,7 @@ public void setItemLabelGenerator(
         Objects.requireNonNull(itemLabelGenerator,
                 "The item label generator can not be null");
         this.itemLabelGenerator = itemLabelGenerator;
-        reset();
+        refreshCheckboxes();
```

Comment:
All the tests pass even without this addition. Could add a Junit test which also checks that the checkbox labels are updated.

Reply:
I tried again locally, seems to break `CheckboxGroupTest.selectItem_setItemLabelGenerator_selectionIsRetained` for me. I wonder why that could happen.

Reply:
I also added an integration test that checks if the labels are updated.

Reply:
> I tried again locally, seems to break

I meant the added line `+ refreshCheckboxes();`, not the entire change. The tests passed with `reset()` removed only.

Reply:
Oh, I get it. Then I suppose the new test addresses this concern.

Reply:
nit: an integration test might be a bit overkill for the purpose. Tried the following and it seems to cover it:

```java
Checkbox cb = (Checkbox) checkboxGroup.getChildren().findFirst().get();
Assert.assertEquals("foo (Updated)", cb.getLabel());
```

Reply:
I updated the PR to address this